tfe/resource_tfe_team_token.go

```diff
@@ -0,0 +1,84 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceTFETeamToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFETeamTokenCreate,
+		Read:   resourceTFETeamTokenRead,
+		Delete: resourceTFETeamTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"team_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"token": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceTFETeamTokenCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get the team ID.
+	teamID := d.Get("team_id").(string)
+
+	log.Printf("[DEBUG] Create new token for team: %s", teamID)
+	token, err := tfeClient.TeamTokens.Generate(ctx, teamID)
+	if err != nil {
+		return fmt.Errorf(
+			"Error creating new token for team %s: %v", teamID, err)
+	}
+
+	d.Set("token", token.Token)
```

Comment:
(as above) can we add a comment to explain why this is here and not in the Read

Reply:
Yep... Fixed in next commit.